Sources/TuistLoader/Loaders/ManifestModelConverter.swift

```diff
@@ -63,7 +63,12 @@ public final class ManifestModelConverter: ManifestModelConverting {
         externalDependencies: [String: [XcodeGraph.TargetDependency]],
         type: XcodeGraph.ProjectType
     ) async throws -> XcodeGraph.Project {
-        let rootDirectory: AbsolutePath = try await rootDirectoryLocator.locate(from: path)
+        let rootDirectory: AbsolutePath = switch type {
+        case .local:
+            try await rootDirectoryLocator.locate(from: path)
+        case .external:
+            (try? await rootDirectoryLocator.locate(from: path)) ?? path
```

Comment:
I'm not sure this is the right place to solve the issue. The logic for what a root directory is is defined inside [`rootDirectoryLocator.locate`](https://github.com/tuist/tuist/blob/b5eb31100e63e85e5a03a0cd6dad69aecf6be6b8/Sources/TuistCore/Utils/RootDirectoryLocator.swift#L60) so I wonder if we should add a last case there where if we detect a `Package.swift` after having tried the other options, we can consider than one the root's directory. Thoughts @fortmarek ?

Reply:
I'm wondering if we should actually keep the current requirement of using `.git` or the `Tuist` directory. `Package.swift` does not always mean you're located in the root directory, for example, when you run `tuist generate` in one of the subdirectories of your project that contains a `Package.swift`, we would consider it to be the root even if it's not.

In other words, using `Package.swift` to find a root of the project is the same as if we used `Project.swift` to find a root.

I think finding the `Tuist.swift` file is the most bulletproof way to find a root of a Tuist project. @alpanyukov, is there a reason why you can't include the file? Would improving the error message that would explain you need to add a `Tuist.swift` help here?

Reply:
A pluggable local spm package should not know that it has been plugged into a tuist project. It remains a regular spm package. For example, such a package could be developed by another team in a monorepository (not .git) that does not use tuist. Why is it necessary to disallow native local spm packages?

Reply:
> when you run tuist generate in one of the subdirectories of your project that contains a Package.swift, we would consider it to be the root even if it's not.

That's why I made this edit here and with a check to an external project

Reply:
Now it turns out that if the spm package to be connected does not have its own Tuist directory and lies in subdirectories relative to the root project - then Tuist from the root project to which we connect will be used for such a package. 
If we connect a package that is higher up the tree and also has no Tuist directory, we get an error.
I think these are equivalent cases of spm package connection. The only difference is its location. But in the first case the project is generated, and in the second case the error occurs.

Reply:
Maybe in case of connecting local spm packages it is possible to make rootDirectory optional?

Reply:
@pepicrft @fortmarek can we find a solution where we don't have to patch a third party local spm package to add it to the dependencies?

Reply:
I think I misunderstood the issue.

So, your structure is the following:
```
// repo-one
- Project.swift -> links to a local package from `repo-two`

// repo-two (completely different directory)
```

Is that correct?

Reply:
Yes! 
Аnd `Package.swift` from `repo-two` may have a dependency on a local `repo-three` package from another directory, and so on.

Reply:
Got it – then I'd follow @pepicrft's [suggestion](https://github.com/tuist/tuist/pull/7156/files#r1882133808) to fallback to using `path` iff the `path` from which we try to find the root has a `Package.swift` file – instead of ignoring any error from the `RootDirectoryLocator`.

Reply:
Great!
Do I need to leave the test and everything related to it in this pull request?
Happy to make edits and post a draft!

Reply:
Let's keep the current PR open. I think a unit test might be enough to test this scenario, so I'd be for removing the fixture and the acceptance test.

You might be able to update `RootDirectoryLocator` and `RootDirectoryLocatorTests` without changing anything else.